internal/page/donor/make_a_new_lpa.go

```diff
@@ -0,0 +1,66 @@
+package donor
+
+import (
+	"net/http"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/actor"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/date"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/page"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/place"
+	"github.com/ministryofjustice/opg-modernising-lpa/internal/validation"
+)
+
+type MakeANewLPAData struct {
+	App         page.AppData
+	Errors      validation.List
+	FullName    string
+	DateOfBirth date.Date
+	Address     place.Address
+}
+
+func MakeANewLPA(tmpl template.Template, donorStore DonorStore) Handler {
+	return func(appData page.AppData, w http.ResponseWriter, r *http.Request, donor *actor.DonorProvidedDetails) error {
+		previouslyProvidedDetails, err := donorStore.Latest(r.Context())
+		if err != nil {
+			return err
+		}
+
+		data := MakeANewLPAData{
+			App:         appData,
+			FullName:    previouslyProvidedDetails.Donor.FullName(),
+			DateOfBirth: previouslyProvidedDetails.Donor.DateOfBirth,
+			Address:     previouslyProvidedDetails.Donor.Address,
+		}
+
+		if data.FullName != donor.Donor.FullName() && donor.Donor.FirstNames != "" {
+			data.FullName = donor.Donor.FullName()
+		} else {
```

Comment:
I may be reading this wrong, but I don't think the `else`s do anything and you don't need the `donorStore.Put`

Reply:
which would also mean you could move the `if POST { }` to the top